pacific_rim_library/indexer.py

```diff
@@ -579,6 +593,9 @@ def download_thumbnail(self, prl_solr_document: PRLSolrDocument):
     def upload_thumbnail(self, prl_solr_document: PRLSolrDocument, filepath: str):
         """Puts the thumbnail on S3."""
 
+        # Determine a URL for the thumbnail now that we've downloaded it and know the image format
+        prl_solr_document.add_thumbnail_url()
```

Comment:
Now, only add a thumbnail URL to a Solr doc if we're going to put that thumbnail in S3. This removes the need for the `discard_incorrect_thumbnail_url` method, because an incorrect thumbnail URL cannot possibly be added to the doc.